### PR TITLE
Create add proponent field

### PR DESCRIPTION
**Pull request by djnunez-aot**
#1631 

-Add proponent field 

-Fetch proponents by ids

### CODE REVIEW

File: epictrack-web/src/components/project/ProjectList.tsx [hunk 1]
```diff
@@ -50,6 +52,21 @@ const ProjectList = () => {
...
   const projects = React.useMemo(() => ctx.data as Project[], [ctx.data]);
 
+  async function fetchProponents() {
+    const project_proponents = projects
+      .map((p) => p.proponent_id)
+      .filter((ele, index, arr) => arr.findIndex((t) => t === ele) === index);
+
+    const proponentsResponses = await Promise.all(
+      project_proponents.map((proponentId) =>
+        proponentService.getById(proponentId.toString())
```
Comment (jadmnsaad) [inline epictrack-web/src/components/project/ProjectList.tsx]: can we try to find another solution for this other than doing multiple API Calls, one for each proponent Id, can we fetch the whole list from the backend?

File: epictrack-web/src/components/project/ProjectList.tsx [hunk 2]
```diff
@@ -50,6 +52,21 @@ const ProjectList = () => {
 
   const projects = React.useMemo(() => ctx.data as Project[], [ctx.data]);
 
+  async function fetchProponents() {
+    const project_proponents = projects
+      .map((p) => p.proponent_id)
```
Comment (jadmnsaad) [inline epictrack-web/src/components/project/ProjectList.tsx]: let's try using full words for variables please for any new implementation

Comment (djnunez-aot) [reply epictrack-web/src/components/project/ProjectList.tsx]: sounds good

File: epictrack-web/src/components/project/ProjectList.tsx (hunk 1 above)
Comment (djnunez-aot) [reply epictrack-web/src/components/project/ProjectList.tsx]: yea ill do that then ill create a backend route to get all the ids

**Comment by dinesh-aot** [comment]
@djnunez-aot  We already have proponent info coming from project api response. Make use of that like rest of the column filter and place it between SubType and ENV Region fields. You don't need any back end calls in this ticket. Thanks

**Review by dinesh-aot — APPROVED** [review]

**Pull request merged**